.github/workflows/misspell.yaml

```
@@ -0,0 +1,31 @@
+name: misspell
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - release-*
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  misspell:
+    name: misspell
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: misspell
+        uses: reviewdog/action-misspell@v1
```

```suggestion
      - name: misspell
        uses: reviewdog/action-misspell@v1
```

what about we just add these lines to code-spell CI check so that we don't have one more extra GitHub action?